src/filesystem/impls/appshell/node/FileWatcherDomain.js

```diff
@@ -104,41 +65,56 @@ function unwatchPath(path) {
 }
 
 /**
+ * Transform Node's native fs.stats to a format that can be sent through domain
+ * @param {stats} Node's fs.stats result
+ * @return {object} Can be consumed by new FileSystemStats(object); in Brackets
+ */
+function normalizeStats(nodeFsStats) {
+    // from shell: If "filename" is a symlink,
+    // realPath should be the actual path to the linked object
+    // not implemented in shell yet
+    return {
+        isFile: nodeFsStats.isFile(),
+        isDirectory: nodeFsStats.isDirectory(),
+        mtime: nodeFsStats.mtime,
+        size: nodeFsStats.size,
+        realPath: null,
+        hash: nodeFsStats.mtime.getTime()
+    };
+}
+
+/**
  * Watch a file or directory.
  * @param {string} path File or directory to watch.
+ * @param {array} ignored List of File or directory to NOT watch.
  */
-function watchPath(path) {
+function watchPath(path, ignored) {
     if (_watcherMap.hasOwnProperty(path)) {
         return;
     }
 
     try {
-        var watcher;
-
-        if (fsevents) {
-            watcher = fsevents(path);
-            watcher.on("change", function (filename, info) {
-                var parent = filename && (fspath.dirname(filename) + "/"),
-                    name = filename && fspath.basename(filename),
-                    type;
-
-                switch (info.event) {
-                case "modified":    // triggered by file content changes
-                case "unknown":     // triggered by metatdata-only changes
-                    type = "change";
-                    break;
-                default:
-                    type = "rename";
-                }
-
-                _domainManager.emitEvent("fileWatcher", "change", [parent, type, name]);
-            });
-        } else {
-            watcher = fs.watch(path, {persistent: false}, function (event, filename) {
-                // File/directory changes are emitted as "change" events on the fileWatcher domain.
-                _domainManager.emitEvent("fileWatcher", "change", [path, event, filename]);
-            });
-        }
+        var watcher = chokidar.watch(path, {
+            persistent: true,
+            ignoreInitial: true,
+            ignorePermissionErrors: true,
+            followSymlinks: true,
+            ignored: ignored,
+            usePolling: process.platform === "win32"
+        });
+
+        watcher.on("all", function (event, filename, nodeFsStats) {
+            if (event === "raw" || event === "error" || !filename) {
```

Comment:
If I'm not mistaken, before when there was a filename an event was always emitted.
Are this change intentional?